lib/release_notes_bot_web/views/slack_interaction_view.ex

```diff
@@ -0,0 +1,70 @@
+defmodule ReleaseNotesBotWeb.SlackInteractionView do
+  use ReleaseNotesBotWeb, :view
+
+  @channel_config_event "channel-configured-with-client"
+  @new_project_event "new-project"
+  @new_client_event "new-client"
+  @manual_release_event "manual-release"
+
+  def message(@manual_release_event, user, project, details) do
+    "<!here>\n#{user} has posted a Release Note to '#{project}' titled: '#{details.title}'.\nDetails:\n#{details.message}\n\nView on Confluence: #{details.persistence_url}"
+  end
+
+  def message(@new_project_event, user, client, project, url, webhook) do
+    "#{user} has created a new project under #{client} titled: '#{project}'. Release notes will be persisted to #{where_to_persist(url) <> determine_serve_repo_webhook_url(webhook)}"
+  end
+
+  def message(@new_client_event, user, client) do
+    "#{user} has created new client: #{client}"
+  end
+
+  def message(@channel_config_event, user, client) do
+    "#{user} has configured this channel to accept messages and updates for projects under: #{client}"
+  end
+
+  defp where_to_persist(url) do
+    case url do
+      nil -> "the default location."
+      _ -> "<#{url}|this specified location>."
+    end
+  end
+
+  defp format_repo_instructions(content) do
+    content
+    # split on numbers in list
+    |> String.split(~r/(?<!\S)\d(?![^\s.,?!])/)
+    # get the 5th item in the list - the instructions we want
+    |> Enum.at(5)
+    # split on new lines
+    |> String.split("\n")
+    # drop the first item in the list
+    |> Enum.drop(1)
+    # drop the last two items in the list
+    |> Enum.drop(-2)
+    # reduce the list into a string
+    |> Enum.reduce("", fn line, acc -> acc <> line <> "\n" end)
+    # replace the dashes with bullets for slack formatting
+    |> String.replace(["- "], "• ")
```

Comment:
this is gross. Is this just being formatted for `slack`?

Reply:
I think I would prefer that the stings get duplicated, as long as it was clear what text was being presented.